Makefile

```diff
@@ -271,14 +271,46 @@ test.e2e:
 # Operations - Local Deployment
 # ------------------------------------------------------------------------------
 
-run: manifests generate fmt vet ## Run a controller from your host.
-	go run ./internal/cmd/main.go
+# NOTE: the environment used for "make debug" or "make run" by default should
+#       have a Kong Gateway deployed into the kong-system namespace, but these
+#       defaults can be changed using the arguments below.
+#
+#       One easy way to create a testing/debugging environment that works with
+#       these defaults is to use the Kong Kubernetes Testing Framework (KTF):
+#
+#       $ ktf envs create --addon metallb --addon kong --kong-disable-controller --kong-admin-service-loadbalancer
+#
+#       KTF can be installed by following the instructions at:
+#
+#       https://github.com/kong/kubernetes-testing-framework
+
+KUBECONFIG ?= "${HOME}/.kube/config"
+KONG_NAMESPACE ?= kong-system
+KONG_PROXY_SERVICE ?= ingress-controller-kong-proxy
+KONG_ADMIN_PORT ?= 8001
+KONG_ADMIN_URL ?= "http://$(shell kubectl -n kong-system get service ingress-controller-kong-admin -o=go-template='{{range .status.loadBalancer.ingress}}{{.ip}}{{end}}'):$(KONG_ADMIN_PORT)"
```

Comment:
Does this work with multiple ingress service IPs?

Probably still good enough for the cases these are usually used even if not, so just a comment rather than a change request.

Reply:
yeah it wont, it's kinda custom built around ktf, but that's still better than the previous one that didn't work at all I think.